src/app/examples/form/form.actions.ts

```diff
@@ -0,0 +1,33 @@
+import { Action } from '@ngrx/store';
+import { HttpErrorResponse } from '@angular/common/http';
+
+import { Form } from '@app/examples/form/form.model';
+
+export enum FormActionTypes {
+  RETRIEVE = '[Form] Retrieve',
+  RETRIEVE_SUCCESS = '[Form] Retrieve Success',
+  RETRIEVE_ERROR = '[Form] Retrieve Error'
+}
+
+export class ActionFormRetrieve implements Action {
+  readonly type = FormActionTypes.RETRIEVE;
+
+  constructor(readonly payload: any) {}
+}
+
+export class ActionFormRetrieveSuccess implements Action {
+  readonly type = FormActionTypes.RETRIEVE_SUCCESS;
+
+  constructor(readonly payload: { form: any }) {}
+}
+
+export class ActionFormRetrieveError implements Action {
+  readonly type = FormActionTypes.RETRIEVE_ERROR;
+
+  constructor(readonly payload: { error: HttpErrorResponse }) {}
```

Comment:
As previously mentioned, this works only with local storage so do we need `HttpErrorResponse` ? On the other hand, is there some backend where we could do real `http` requests ?